### ISSUE TITLE
Nodes reappear temporarily when digging

### ISSUE THREAD

**Issue by HybridDog** [minetest/minetest#7098]
##### Issue type

 - Bug report
 
##### Summary

When digging multiple nodes with a fast tool, the nodes disappear, reappear and then disappear again. This defeats the purpose of very fast tools, especially when multiple nodes temporarily reappear.

I guess something like this happens:
* The player digs a node at pos1, the action is sent to the server. client predicts air at pos1
* The server removes the node at pos1.
* The player digs a node at pos2, the action is sent to the server. client predicts air at pos2
* The server sends the changed mapblock resulting from the first digging.
* The client assumes that the received mapblock is more up to date than the prediction, thus the node at pos2 is set back to what it was before, the air at pos1 is not changed.
* The server removes the node at pos2.
* The server sends the changed mapblock.
* The client (now correctly) assumes that the received mapblock is more up to date than the prediction, thus the node at pos2 is set to air again.

Note that l don't know whether l guessed correctly.

##### Steps to reproduce

play on some server with a very fast pick, hold LMB to dig multiple nodes, you should then notice that you're digging the same nodes multiple times (from client perspective)

**Comment by paramat** [comment]
But if the tool wear is serverside this comes at no disadvantage?
Seems expected, a minor issue and non-trivial to change.

**Comment by rubenwardy** [comment]
To fix this, you'd need to have the server acknowledge any digs. Any unacknowledged digs are then replayed

**Comment by azekillDIABLO** [comment]
I agree this bug is reallu pesky, I often use pickaxes that mine faster that I fall but everytime blocks reapear and bump me. I would also tell it does for blocks you've placed: they tend to disappear just after you've placed them for a short moment. A nightmare for skyblock.

**Comment by HybridDog** [comment]
@paramat If this also happens when placing nodes (needs testing, with and without anticheat), it's not low priority because placing a way you're walking on using sneak becomes impossible.

**Comment by HybridDog** [comment]
> I think I've found a way to strongly mitigate this problem: Send single node changes instead of whole mapblocks.

This approach probably didn't work correctly because the server does not know which version of the map the client has and mapblocks are often sent for some other reason than the player's interacting.

#

Here's a video of the problem when placing (not digging) nodes:
https://github.com/minetest/minetest/assets/3192173/180e2c19-300c-476b-99f0-9fba5981cf77

To make the video, I have simulated a slow network connection on Linux:
```
sudo tc qdisc add dev lo root handle 1: htb default 12
sudo tc class add dev lo parent 1:1 classid 1:12 htb rate 16kbps ceil 32kbps
sudo tc qdisc add dev lo root netem delay 600ms
```
